pex/common.py

```diff
@@ -598,7 +633,7 @@ def zip(
         if labels:
             selected_files = set(
                 itertools.chain.from_iterable(self.filesets.get(label, ()) for label in labels)
-            )
+            )  # type: Iterable[str]
```

Comment:
This code actually cares about the underlying setness in reality of the types in both branches since zips will happily store a given filename more than once and we don't want that.

Reply:
Isn't it more deterministic if we allow that?

Reply:
As I said above - I think the whole dup business is fake and there are no dups in practice. But if there were I'd want PEX to error unless the file contents were identical (which PEX checks and does elsewhere - for example venv creation). My main point is, in the small, the current Chroot code permits dups and has certain behavior wrt to them. That behavior should not change unless your point is very deliberately to do that. I think that's not the point of your change though if I read correctly.

Reply:
Ok, great! I have vastly restricted this change to only cover the quadratic behavior when searching for whether a file already exists.